idaes/power_generation/costing/tests/test_NGCC_costing.py

```diff
@@ -0,0 +1,352 @@
+##############################################################################
+# Institute for the Design of Advanced Energy Systems Process Systems
+# Engineering Framework (IDAES PSE Framework) Copyright (c) 2018-2020, by the
+# software owners: The Regents of the University of California, through
+# Lawrence Berkeley National Laboratory,  National Technology & Engineering
+# Solutions of Sandia, LLC, Carnegie Mellon University, West Virginia
+# University Research Corporation, et al. All rights reserved.
+#
+# Please see the files COPYRIGHT.txt and LICENSE.txt for full copyright and
+# license information, respectively. Both files are also available online
+# at the URL "https://github.com/IDAES/idaes-pse".
+##############################################################################
+"""
+Case B31A - Natural Gas Combined Cycle (NGCC) plant_gross_power
+Reference: NETL Baseline Report Rev 4
+Author: A. Deshpande and M. Zamarripa
+"""
+import pytest
+from idaes.power_generation.costing.power_plant_costing import \
+     (get_PP_costing,
+      build_flowsheet_cost_constraint,
+      costing_initialization)
+from idaes.core.util.model_statistics import degrees_of_freedom
+import pyomo.environ as pyo
+from idaes.core import FlowsheetBlock
+from idaes.core.util import get_solver
+
+# Get default solver for testing
+solver = get_solver()
+
+
+@pytest.mark.unit
+def test_get_costing():
+    # Create a Concrete Model as the top level object
+    m = pyo.ConcreteModel()
+
+    # Add a flowsheet object to the model
+    m.fs = FlowsheetBlock(default={"dynamic": False})
+    m.fs.get_costing(year='2018')
+
+    # Accounts with Feedwater Flow to HP section of HRSG, as the
+    # reference/scaling parameter - Exhibit 5-15
+    FW_accounts = ['3.1', '3.3', '8.4']
+
+    m.fs.unit = pyo.Block()
+    m.fs.unit.feedwater_flowrate = pyo.Var(initialize=1085751)  # lb/hr
+    m.fs.unit.feedwater_flowrate.fix()
+    get_PP_costing(m.fs.unit, FW_accounts,
+                   m.fs.unit.feedwater_flowrate, 'lb/hr', 6)
+    assert isinstance(m.fs.unit.costing.total_plant_cost, pyo.Var)
+    assert hasattr(m.fs.unit.costing, "bare_erected_cost")
+    assert isinstance(m.fs.unit.costing.total_plant_cost_eq, pyo.Constraint)
+    assert degrees_of_freedom(m) == 0
+
+
+@pytest.mark.solver
+@pytest.mark.skipif(solver is None, reason="Solver not available")
+@pytest.mark.component
+def test_get_ngcc_costing():
```

Comment:
This is one huge test! Is there anyway to break it up into smaller pieces that can be checked separately?